src/coreclr/jit/helperexpansion.cpp

```diff
@@ -610,6 +613,14 @@ bool Compiler::fgExpandThreadLocalAccessForCallNativeAOT(BasicBlock** pBlock, St
         GenTree* tlsRootOffset = gtNewIconNode((size_t)tlsRootObject, TYP_INT);
         tlsRootOffset->gtFlags |= GTF_ICON_SECREL_OFFSET;
 
+        if (TargetArchitecture::IsArm64)
+        {
+            // for windows/arm64, the immediate constant should be contained because it gets
+            // generated as part of ADD instruction that consumes this constant. See
+            // emitIns_Add_Add_Tls_Reloc().
+            tlsRootOffset->SetContained();
+        }
```

Comment:
This expansion runs way before lower, even before we have LIR... It' odd to be setting things as contained at this point. Can it be done in lowering instead? What would happen if some phase decided to break apart this constant and its parent between this point and lowering?

Reply:
Yeah, I was not entirely sure the right place to set this, but probably should be able to do it in lowering.